examples/platform/silabs/wifi/icd/vendor-handlers/AppleKeychainHandler.h

```diff
@@ -44,15 +46,26 @@ class AppleKeychainHandler : public VendorHandler<AppleKeychainHandler>
     {
         VerifyOrReturnValue(subscriptionsInfoProvider != nullptr && fabricTable != nullptr, false);
 
-        bool hasValidException = false;
+        bool hasValidException = true; // Default to true if no VendorId::Apple fabric is found
 
-        for (auto it = fabricTable->begin(); it != fabricTable->end(); ++it)
+        if (fabricTable->FabricCount() == 0)
```

Comment:
Could be `VerifyOrReturnValue(fabricTable->FabricCount(), false)`

Reply:
updated